dali/python/nvidia/dali/data_node.py

```diff
@@ -122,6 +139,63 @@ def __bool__(self):
                 " represent computations in DALI Pipeline see the \"Mathematical Expressions\""
                 " section of DALI documentation."))
 
+    def __getitem__(self, val):
+        idxs = []
+        new_axes = []
+        new_axis_names = []
+
+        # returns True if this index adds a new output dimension
+        def process_index(idx, dim):
```

Comment:
I think we need a lot of type checking in this function. This is Python, so we can and will get stuff like DataNode()[DataNode()], DataNode()[np.array(...)], DataNode()["duck"].

I think we are limiting to integers now, right?

Reply:
Now, we're not and we do support datanodes and slices with datanodes and a mix of datanodes and integers.

Reply:
Ack. Didn't think about Argument Inputs. Just wonder how the errors would look if we use a float scalar here for example.